src/_langs/en/updates/2015-07-22-measuring-performance-in-a-service-worker.markdown

```diff
@@ -0,0 +1,162 @@
+the requests that are made in the context of the worker. If you can only see 
+what happens from the context of the current window you would miss critical 
+timing information from:
+
+* `fetch()` requests made inside the `oninstall` event of the service worker
+* `fetch()` requests made when caching data in an `onpush` event can now be 
+  traced to understand the performance your users see.
+* finally, `fetch()` request that are made and intercepted by the `onfetch` 
+  handler.
+
+The last point is important. One way to think of a Service Worker is as a proxy 
+that sits between the web UI and the network.  The `performance` object on the 
+`window` only sees the timings and information for the part of the request that it 
+invokes, it has no knowledge of the Service Worker sitting in between the client 
+and the network so it can't understand the impact of the service worker.
+
+## How can I use this?
+
+A typical service worker that support offline first would have an 
+[install step where it will download and save all the assets for later use](https://github.com/GoogleChrome/voice-memos/blob/27f2ceac81aa2ee1a02fbce05bedce571709b29e/src/scripts/sw.js)
+
+An example of where this could be used is to record and log the timing data of 
+the install step so you can make some measured decisions about how you can improve
+the performance of your install based on real in the wild user usage.
+
+{% highlight javascript %}
+self.addEventListener("install", function() {
+  var urls = [
+    '/',
+    '/images/chrome-touch-icon-192x192.png',
+    '/images/ic_add_24px.svg',
+    '/images/side-nav-bg@2x.jpg',
+    '/images/superfail.svg',
+    '/scripts/voicememo-core.js',
+    '/styles/voicememo-core.css',
+    '/third_party/Recorderjs/recorder.js',
+    '/third_party/Recorderjs/recorderWorker.js',
+    '/third_party/Recorderjs/wavepcm.js',
+    '/third_party/moment.min.js',
+    '/favicon.ico',
+    '/manifest.json'
+  ];
+
+  urls = urls.map(function(url) {
+    return new Request(url, {credentials: 'include'});
+  });
+
+  event.waitUntil(
+    caches
+      .open(CACHE_NAME + '-v' + CACHE_VERSION)
+      .then(function(cache) {
+        // Fetch all the URL's and store in the cache
+        return cache.addAll(urls);
+      })
+      .then(function () {
+        // Analyse all the requests
+        var requests = window.performance.getEntriesByType("resource");
+        
+        // Loop across all the requests and save the timing data.
+        
+        return;
+      })
+  );
+};
+
+});
+{% endhighlight %}
+
+Today many sites use [RUM](https://en.wikipedia.org/wiki/Real_user_monitoring) for understanding
+how the majority of their users experience their site.  Tools like Google Analytics
+already [report site speed data](https://support.google.com/analytics/answer/1205784?hl=en) using the Navigation
+Timing API but will need to be updated to include performance analysis from a Worker context.
```

Comment:
It may be worth clarifying that within the worker _everything_ looks like a resource requests. As in, there are no plans to expose Nav Timing API to Workers; a nav request is like any other resource request.